### PR TITLE
Fix SDcard size and format issue

### LINKED ISSUE #4924: Qx7  report wrong sd card size
If a micro SD card bigger than 4Gig is use, 8Gig etc. The Q X7 information screen report a max of 3800M and refuse to format it, returning a cd card error. 

The card can be formatted on a pc and use as a normal 4Gig card. 

I tested the cards with h2testw and they are fine.

### PR DESCRIPTION
**Pull request by 3djc**
- fix overflow in SDcard size computation
- fix format function
- implement more informative format error handling

This fixes #4924

### CODE REVIEW

**Comment by 3djc** [comment]
I have left error message untranslatable, as users should not see them, and when/if it comes to  us I would prefer it in english, but I'm opened to different opinions

File: radio/src/sdcard.cpp [hunk 1]
```diff
@@ -472,7 +472,7 @@ uint32_t sdGetNoSectors()
 
 uint32_t sdGetSize()
 {
-  return (sdGetNoSectors() * BLOCK_SIZE) / 1000000;
```
Comment (bsongis) [inline radio/src/sdcard.cpp:475]: 👍

File: radio/src/gui/common/stdlcd/radio_sdmanager.cpp [hunk 2]
```diff
@@ -175,12 +175,32 @@ void menuRadioSdManager(event_t _event)
     audioQueue.stopSD();
 #endif
     BYTE work[_MAX_SS];
-    if (f_mkfs(0, FM_FAT32, 0, work, sizeof(work)) == FR_OK) {
-      f_chdir("/");
-      REFRESH_FILES();
-    }
-    else {
-      POPUP_WARNING(STR_SDCARD_ERROR);
+    FRESULT res = f_mkfs("", FM_FAT32, 0, work, sizeof(work));
```
Comment (bsongis) [inline radio/src/gui/common/stdlcd/radio_sdmanager.cpp]: This code is common to all ARM platforms, perhaps it's possible to add a function and avoid redundant code?

Comment (bsongis) [reply radio/src/gui/common/stdlcd/radio_sdmanager.cpp]: ... at least the code which deals with error codes

**Comment by bsongis** [comment]
Thanks!

**Pull request merged**